src/integrationTest/java/org/opensearch/test/framework/ldap/LdapServer.java

```diff
@@ -0,0 +1,250 @@
+		if (isStarted.get()) {
+			throw new IllegalStateException(LdapServer.SERVER_ALREADY_STARTED);
+		}
+		configureAndStartServer(ldifData);
+	}
+
+	private Collection<InMemoryListenerConfig> getInMemoryListenerConfigs() throws Exception {
+		KeyStore keyStore = createEmptyKeyStore();
+		addLdapCertificatesToKeystore(keyStore);
+		final SSLUtil sslUtil = new SSLUtil(createKeyManager(keyStore), createTrustManagers(keyStore));
+
+		ldapPort = SocketUtils.findAvailableTcpPort();
+		ldapsPort = SocketUtils.findAvailableTcpPort();
+
+		Collection<InMemoryListenerConfig> listenerConfigs = new ArrayList<>();
+		listenerConfigs.add(InMemoryListenerConfig.createLDAPConfig("ldap", null, ldapPort, sslUtil.createSSLSocketFactory()));
+		listenerConfigs.add(InMemoryListenerConfig.createLDAPSConfig("ldaps", ldapsPort, sslUtil.createSSLServerSocketFactory()));
+		return listenerConfigs;
+	}
+
+	private static KeyManager[] createKeyManager(KeyStore keyStore)
+		throws NoSuchAlgorithmException, KeyStoreException, UnrecoverableKeyException {
+		KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+		keyManagerFactory.init(keyStore, null);
+		return keyManagerFactory.getKeyManagers();
+	}
+
+	private static TrustManager[] createTrustManagers(KeyStore keyStore) throws NoSuchAlgorithmException, KeyStoreException {
+		TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+		trustManagerFactory.init(keyStore);
+		return trustManagerFactory.getTrustManagers();
+	}
+
+	private void addLdapCertificatesToKeystore(KeyStore keyStore) throws KeyStoreException {
+		keyStore.setCertificateEntry("trustAnchor", trustAnchor.certificate());
+		keyStore.setKeyEntry("ldap-key", ldapCertificate.getKey(), null, new Certificate[]{ ldapCertificate.certificate()});
+	}
+
+	private static KeyStore createEmptyKeyStore() throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException {
+		KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+		keyStore.load(null);
+		return keyStore;
+	}
+
+	private synchronized void configureAndStartServer(LdifData ldifData) throws Exception {
+		Collection<InMemoryListenerConfig> listenerConfigs = getInMemoryListenerConfigs();
+
+		Schema schema = Schema.getDefaultStandardSchema();
+
+		final String rootObjectDN = ldifData.getRootDistinguishedName();
+		InMemoryDirectoryServerConfig config = new InMemoryDirectoryServerConfig(new DN(rootObjectDN));
+
+		config.setSchema(schema);  //schema can be set on the rootDN too, per javadoc.
+		config.setListenerConfigs(listenerConfigs);
+		config.setEnforceAttributeSyntaxCompliance(false);
+		config.setEnforceSingleStructuralObjectClass(false);
+
+		//config.setLDAPDebugLogHandler(DEBUG_HANDLER);
+		//config.setAccessLogHandler(DEBUG_HANDLER);
+		//config.addAdditionalBindCredentials(configuration.getBindDn(), configuration.getPassword());
```

Comment:
should remove this commented code if no longer needed

Reply:
done, code removed.